client/pages/car-details.jsx

```diff
@@ -1,16 +1,23 @@
 import React from 'react';
-import { Card, Table } from 'react-bootstrap';
-
+import { Card, Modal, Table } from 'react-bootstrap';
+import AddForm from '../components/add-record';
 class CarDetails extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      car: {}
+      car: {},
+      modal: false
     };
     this.makeTable = this.makeTable.bind(this);
+    this.getHistory = this.getHistory.bind(this);
+    this.toggleModal = this.toggleModal.bind(this);
   }
 
   componentDidMount() {
+    this.getHistory();
+  }
+
```

Comment:
You should define another method here to receive the added record so you don't have to re-fetch the whole history again.

```js
addRecord(newRecord) {
  // update your state
}
```

then pass that to the modal form

Reply:
added new method that updates the state with the new record received from the add-record form
![image](https://user-images.githubusercontent.com/48267398/163653327-0f64e927-d322-4a50-a1b1-cf55ad7abd90.png)